deploy/example_catalog/cr-app-cdh632cm.json

```diff
@@ -0,0 +1,415 @@
+                    "path": "/"
+                },
+                "id": "hbase-thrift",
+                "label": {
+                    "name": "HBase Thrift Server"
+                }
+            },
+            {
+                "id": "hive-metastore",
+                "label": {
+                    "name": "HIVE Metastore"
+                }
+            },
+            {
+                "endpoint": {
+                    "isDashboard": true,
+                    "urlScheme": "http",
+                    "port": 10002,
+                    "path": "/"
+                },
+                "id": "hive-server",
+                "label": {
+                    "name": "HIVE Server 2"
+                }
+            },
+            {
+                "endpoint": {
+                    "isDashboard": false,
+                    "port": 50111
+                },
+                "id": "webhcat",
+                "label": {
+                    "name": "WebHCat Server"
+                }
+            },
+            {
+                "endpoint": {
+                    "isDashboard": true,
+                    "urlScheme": "http",
+                    "port": 8888,
+                    "path": "/"
+                },
+                "id": "hue",
+                "label": {
+                    "name": "Hue Console"
+                }
+            },
+            {
+                "endpoint": {
+                    "isDashboard": true,
+                    "urlScheme": "http",
+                    "port": 18088,
+                    "path": "/"
+                },
+                "id": "spark-history",
+                "label": {
+                    "name": "Spark history server"
+                }
+            },
+            {
+                "id": "impala",
+                "label": {
+                    "name": "Impala service"
+                }
+            },
+            {
+                "id": "oozie",
+                "label": {
+                    "name": "OOZIE service"
+                }
+            }
+        ],
+        "defaultConfigPackage":  {
+            "packageURL": "file:///opt/configscripts/appconfig.tgz"
+        },
+        "roles": [
+            {
+                "id": "cmserver",
+                "cardinality": "1"
+            },
+            {
+                "id": "primary",
+                "cardinality": "1"
+            },
+            {
+                "id": "hbase",
+                "cardinality": "1"
+            },
+            {
+                "id": "hive",
+                "cardinality": "1"
+            },
+            {
+                "id": "impala",
+                "cardinality": "1"
+            },
+            {
+                "id": "apps",
+                "cardinality": "1"
+            },
+            {
+                "id": "worker",
+                "cardinality": "3+"
+            },
+            {
+                "id": "gateway",
+                "cardinality": "0+"
```

Comment:
Are all these roles required. With this we need 9 pods to bring up CDH?

Reply:
@swamibluedata we have tried to be as close to CDH 6.x native EPIC app as possible with this one. The json for that can be found [here](https://github.com/bluedatainc/bluedata-catalog/blob/cdh610_templates-master/cdh610.json). The native EPIC app requires 9 containers and customers love it :)

@kmathur2 and tatsuya can provide more reasoning behind this design.

Reply:
I can vouch for the workers having cardinality as 3+ because without min 3 datanodes CDH throws a lot of warnings.

Reply:
👍

Reply:
Yes, once we separated the services on different roles the requirement for the number of initial containers/pods increased. However, the min CPU and memory requirement could be low, so we will be using same amount of resources just spread across pods.

Reply:
@swamibluedata Made a change to have cardinality as 0+ for hbase, impala and apps roles. This way we will need minimum 6 pods( 1 cmserver, 1 primary, 1 hive , 3 workers) for bringing up CDH cluster.

@kmathur2 The minimum memory and CPU requirements for roles are in accordance with min_cores and min_memory requirements in EPIC.